src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js

```diff
@@ -27,247 +28,228 @@ import RequiredValueErrorMessage from './required-value-error-message';
-                      <LanguagesControl
-                        isClosed={true}
-                        onClick={() => {
-                          // expand all multiline language inputs in case the
-                          // first one was expanded when all languages
-                          // are shown
-                          if (
-                            this.state.expandedTranslations[
-                              this.props.selectedLanguage
-                            ]
-                          ) {
-                            this.expandAllTranslations();
-                          }
-                          this.toggleLanguages();
-                        }}
-                        remainingLanguages={languages.length - 1}
-                      />
-                    );
-                  if (isLastLanguage)
-                    return (
-                      <LanguagesControl
-                        onClick={this.toggleLanguages}
-                        remainingLanguages={languages.length - 1}
-                        isDisabled={Boolean(
-                          this.props.hasError ||
-                            hasErrorOnRemainingLanguages ||
-                            this.props.hasWarning ||
-                            hasWarningOnRemainingLanguages
-                        )}
-                      />
-                    );
-                  return null;
-                })()}
-                hasError={Boolean(
-                  this.props.hasError ||
-                    (this.props.errors && this.props.errors[language])
-                )}
-                hasWarning={Boolean(
-                  this.props.hasWarning ||
-                    (this.props.warnings && this.props.warnings[language])
-                )}
-                intl={this.props.intl}
-                warning={this.props.warnings && this.props.warnings[language]}
-                error={this.props.errors && this.props.errors[language]}
-                {...createLocalizedDataAttributes(this.props, language)}
-              />
-            );
-          })}
-        </Spacings.Stack>
-      </Constraints.Horizontal>
     );
+    setExpandedTranslations(newExpandedTranslations);
+  }, [expandedTranslations]);
+
+  const languages = sortLanguages(
+    props.selectedLanguage,
+    Object.keys(props.value)
+  );
+
+  const hasErrorInRemainingLanguages =
+    props.hasError ||
+    getHasErrorOnRemainingLanguages(props.errors, props.selectedLanguage);
+  const hasWarningInRemainingLanguages =
+    props.hasWarning ||
+    getHasWarningOnRemainingLanguages(props.warnings, props.selectedLanguage);
+
+  if (hasErrorInRemainingLanguages || hasWarningInRemainingLanguages) {
+    if (!areLanguagesOpened) {
+      // this update within render replaces the old `getDerivedStateFromProps` functionality
+      // https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops
+      toggleLanguages();
```

Comment:
I find this waaaaaaaaaaaaaaaaay easier to read than the old `getDerivedStateFromProps`